python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py

```diff
@@ -410,7 +410,7 @@ def get_vrf_router_id(
                 )
                 raise AristaAvdInvalidInputsError(msg)
             # Resolve router ID from loopback interface
-            return get_ip_from_ip_prefix(interface_data["ip_address"])
+            return get_ip_from_ip_prefix(str(interface_data.ip_address))
```

Comment:
I suppose the type error was that the `ip_address` field may be `None`. I think it would be better to just add an `or not interface_data.ip_address` to the `if` above, which means we _know_ it is set by now, so you can remove the `str()`

Reply:
Fixed!